content/articles/how-to-add-support-for-spatial-data-in-your-database-using-geoalchemy-and-spatialite/index.md

```diff
@@ -0,0 +1,396 @@
+```
+___
+_`Spatialite` is an sqlite extension so this extension works for sqlite only, if you're using a different database like PostgresQL, you need to install the extension for it too (in which case is called PostGIS). However, you may not need to load the extension this way with other databases too. This is just for demonstration purposes on how this could be done in a development environment._
+___
+
+Lastly, we need to get a later version of sqlite3 installed because the one that comes preinstalled with python, doesn't include 'RTree' which is a datastructure thats used for handling spatial information. So head over to [Sqlite3's website](https://sqlite.org/download.html) and download the latest one (available under Pre-compiled Binaries for Windows). We will replace this with the one in the python binaries. 
+
+Once downloaded, extract the contents and you should find a `sqlite3.dll` file - this is what we need. So copy it and look for the sqlite.dll that comes preinstalled with python, this should be in a folder called `DLLS` that's usually in the same location where python is installed. Mine is found at 
+
+```
+C:\Users\Paulo\AppData\Local\Programs\Python\Python38\DLLs
+```
+once there copy the old one to a safe location as backup, and paste the newly downloaded one in there to replace the old one, it might ask for admin-access so grant it access to copy the file there, and we'll be on our way.
+
+#### <u>Updating Our Models And Creating The Database</u>
+
+Now that we've added the support for spatial data to the database engine, we will now add some special fields to the models from earlier, these are the spatial fields that hold geometric information that can be handled by the database as such. 
+
+We add them with the aid of `GeoAlchemy2` which we have installed earlier on, so go to the `models.py` file and update it so that the two models now contain these new fields that will hold spatial data.
+
+```python
+from geoalchemy2 import Geometry
+
+class User(db.Model):
+    # .. code ommited for brevity
+    # ..
+    geometry = db.Column(Geometry(geometry_type='POINT', management = True, srid='4269'))
+
+class Gig(db.Model):
+    # .. code ommited for brevity
+    # ..
+    gig_geometry = db.Column(Geometry(geometry_type='POINT', management = True, srid='4269'))
+```
+Geoalchemy2 helps us to define these 'geometrical' fields which can have different `geometry_type` as specified by the [WKT](https://en.wikipedia.org/wiki/Well-known_text_representation_of_geometry) (Well Known Text) format. The longitude and latitude values for my application would be in the form of a point, or at least thats what I anticipate:
+
+```
+X(lon, lat)
+```
+Hence I specify the `geometry_type` attribute for the field as `POINT`, this may differ depending on your usecase. You can always check the geolalchemy2 reference docs [here](https://geoalchemy-2.readthedocs.io/en/latest/types.html) to see the available options.
+
+#### <u>**Creating The Database**</u>
```

Comment:
```suggestion
#### Creating The Database
```

Reply:
Thanks for the update @srishilesh i have pushed some updates that addressed the issues pointed out